docs/website/docs/guides/developer-tips.md

```diff
@@ -0,0 +1,416 @@
+    5: 00000000000017f0   207 FUNC    LOCAL  DEFAULT    7 iree_f2h_ieee
+    ...
+```
+
+The `iree-dump-module` tool can also be used to see information about a given
+`.vmfb` file:
+
+```console
+$ iree-dump-module simple_abs.vmfb
+
+//===---------------------------------------------------------------------===//
+// @module : version 0
+//===---------------------------------------------------------------------===//
+
+Required Types:
+  [  0] i32
+  [  1] i64
+  [  2] !hal.allocator
+  [  3] !hal.buffer
+  ...
+
+Module Dependencies:
+  hal, version >= 0, required
+
+Imported Functions:
+  [  0] hal.ex.shared_device() -> (!vm.ref<?>)
+  [  1] hal.allocator.allocate(!vm.ref<?>, i32, i32, i64) -> (!vm.ref<?>)
+  ...
+
+Exported Functions:
+  [  0] abs(!vm.ref<?>) -> (!vm.ref<?>)
+  [  1] __init() -> ()
+
+...
+```
+
+## Dumping executable files
+
+The `--iree-hal-dump-executable-*` flags instruct the compiler to save files
+related to "executable translation" (code generation for a specific hardware
+target) into a directory of your choosing. If you are interested in seeing which
+operations in your input program were fused into a compute kernel or what device
+code was generated for a given program structure, these flags are a great
+starting point.
+
+Flag | Files dumped
+---- | ------------
+`iree-hal-dump-executable-files-to` | All files (meta-flag)
+`iree-hal-dump-executable-sources-to` | Source `.mlir` files prior to HAL compilation
+`iree-hal-dump-executable-intermediates-to` | Intermediate files (e.g. `.o` files, `.mlir` stages)
+`iree-hal-dump-executable-binaries-to` | Binary files (e.g. `.so`, `.spv`, `.ptx`), as used in the `.vmfb`
+`iree-hal-dump-executable-benchmarks-to` | Standalone benchmark files for `iree-benchmark-module`
+
+=== "CPU"
+
+    ```console hl_lines="5 6"
+    $ mkdir -p /tmp/iree/simple_abs/
+
+    $ iree-compile simple_abs.mlir \
+      --iree-hal-target-backends=llvm-cpu \
+      --iree-llvmcpu-link-embedded=false \
+      --iree-hal-dump-executable-files-to=/tmp/iree/simple_abs \
+      -o /tmp/iree/simple_abs/simple_abs_cpu.vmfb
+
+    $ ls /tmp/iree/simple_abs
+
+    module_abs_dispatch_0.mlir
+    module_abs_dispatch_0_system_elf_x86_64_benchmark.mlir
+    module_abs_dispatch_0_system_elf_x86_64.codegen.bc
+    module_abs_dispatch_0_system_elf_x86_64.linked.bc
+    module_abs_dispatch_0_system_elf_x86_64.optimized.bc
+    module_abs_dispatch_0_system_elf_x86_64.s
+    module_abs_dispatch_0_system_elf_x86_64.so
+    simple_abs_cpu.vmfb
+    ```
+
+    !!! tip "Tip - Embedded and system linking"
+
+        The default value of `--iree-llvmcpu-link-embedded=true` generates
+        platform-agnostic ELF files. By disabling that flag, the compiler will
+        produce `.so` files for Linux, `.dll` files for Windows, etc. While ELF
+        files are more portable, inspection of compiled artifacts is easier with
+        platform-specific shared object files.
```

Comment:
I would rephrase this slightly to be less about ELF vs non-ELF container format, and more about platform-standard vs IREE-custom. Because even on platforms like Linux where ELF is standard, IREE's flavor (embedded-elf) is not as thoroughly understood by non-IREE tools. Probably just as a matter of being more aggressively optimized for size. IREE says "symbol table? who needs a symbol table when you have `iree_hal_executable_library_query` ? Of course, a symbol table is useful to use tools such as `objdump` to disassemble symbols, when using profilers, Tracy sampling, etc.

Reply:
Okay, reworded.